src/connect_edges.js

```diff
@@ -91,67 +89,72 @@ function nextPos(pos, resultEvents, processed, origIndex) {
  * @param  {Array.<SweepEvent>} sortedEvents
  * @return {Array.<*>} polygons
  */
-export default function connectEdges(sortedEvents, operation) {
+export default function connectEdges(sortedEvents) {
   let i, len;
   const resultEvents = orderEvents(sortedEvents);
 
   // "false"-filled array
   const processed = {};
   const result = [];
   let event;
+  let depth = [];
+  let holeOf = [];
 
   for (i = 0, len = resultEvents.length; i < len; i++) {
+
     if (processed[i]) continue;
-    const contour = [[]];
-
-    if (!resultEvents[i].isExteriorRing) {
-      if (operation === DIFFERENCE && !resultEvents[i].isSubject && result.length === 0) {
-        result.push(contour);
-      } else if (result.length === 0) {
-        result.push([[contour]]);
-      } else {
-        result[result.length - 1].push(contour[0]);
+    const contour = new Contour();
```

Comment:
Just a note about terminology: In the a paper a "contour" really is just any "linear ring / closed line segment", i.e., it can be either the exterior of polygon or polygon holes:

![image](https://user-images.githubusercontent.com/3620703/73132628-e8ab6f80-401d-11ea-9df4-894e15220ecf.png)

In that sense the class `Contour` is a bit confusing, because it doesn't seem to represent a single ring, but rather an entire polygon. Renaming the class to `Polygon` would be more consistent. But I don't fully understand the semantics of `contourId` yet, is it actually a "polygon id"?

Reply:
I think I get it now -- the contours aren't storing the points of holes like a polygon would, they are merely references to other contour ids.